### ISSUE TITLE
Navigation Menu: handle overflowing Nav Items

### ISSUE THREAD

**Issue by talldan** [WordPress/gutenberg#18298]
Currently when lots of items are added to the navigation menu, the block forces the editor container to widen, which adds a horizontal scrollbar to the entire editor:
![Screen Shot 2019-11-06 at 5 47 50 pm](https://user-images.githubusercontent.com/677833/68287395-9e20f380-00bd-11ea-9e72-fe8b57951efd.png)

It looks kinda terrible!

Instead, we might consider adding a horizontal scroll container within the block itself.

See the table block for prior art on this.

**Comment by getdave** [comment]
cc @mtias because you've mentioned this as an Issue. It's already on the board and we will address it.

**Comment by getdave** [comment]
### Needs decision

There are now two distinct approaches

* [Overflow with scroll](https://github.com/WordPress/gutenberg/pull/18336) - has several UI hacks to make toolbars work.
* [Overflow with wrap on select](https://github.com/WordPress/gutenberg/pull/18429) - solves a number of issues but can feel a little jarring.


[@shaunandrews  has also suggested having a "more" mechanic](https://github.com/WordPress/gutenberg/pull/18429#issuecomment-553551751), but there we a number of edge cases / concerns raised by others around this. 

I've also worked on a mechanic for allowing the parent Block to consume the child Block's toolbar which could help. But I believe @mtias doesn't see this as something the Nav Block will make use of.

I think we're at a point where we need to make a decision on the way forward.

**Comment by jasmussen** [comment]
We have another approach as well: Wrap. https://github.com/WordPress/gutenberg/pull/18431

It wraps it in the editor, and the frontend, keeping the two consistent. It's very little code, and it solves the overflow issue.

**Comment by folletto** [comment]
I think to answer this we need to go back to the principles the Editor is going for. While we aren't aiming for perfect accuracy, we want for the Editor to match as possible the frontend experience. That should be the driving principle for this decision. As such:

* I don't expect any navigation scrolling
* I don't expect any navigation to change layout when clicked

Which means that the only two viable solutions are only:

* Always wrap — most themes do this
* More mechanic — nice, but I consider this for a v2 of this feature

tl;dr: **Always wrap**

**Comment by mtias** [comment]
Just catching up with this conversation. A scrolling container doesn't make a lot of sense because it only "fixes" an editor issue; it doesn't communicate to the user what they are actually creating. In general, wrapping is what I would expect, as it shows the user that their navigation might need some tweaking. In the future, allowing users to opt-in to having all overflowing items grouped within a "more" dropdown could be nice.

The only issue with wrapping is the conflicts it might have with dropdowns, but that is also the case with the front-end. If you have a navigation menu that goes into multiple lines and also items with dropdown in the first line, it's going to be a rough experience for visitors.

tl;dr: I agree with @folletto

**Comment by getdave** [comment]
Thanks @folletto @mtias and @jasmussen. I think that's a definite 👍to moving forward with [@jasmussen's wrap approach](https://github.com/WordPress/gutenberg/pull/18431) and get this Issue resolved. 

@shaunandrews would you be ok to open another Issue to discuss the "More " option as a longer-term solution?

**Comment by karmatosed** [comment]
I also would 👍wrap and the more can come later. For what it's worth unless we have more on front then the idea of 'more' isn't really aligning.

**Comment by shaunandrews** [comment]
> would you be ok to open another Issue to discuss the "More " option as a longer-term solution?

Sure, makes sense to move it to a future iteration.
